pilot/cmd/pilot-agent/status/server.go

```diff
@@ -101,3 +133,38 @@ func (s *Server) handleReadyProbe(w http.ResponseWriter, _ *http.Request) {
 		s.lastProbeSuccessful = true
 	}
 }
+
+func (s *Server) handleAppReadinessProbe(w http.ResponseWriter, req *http.Request) {
+	requestStatusCode(fmt.Sprintf("http://127.0.0.1%s", s.appReadyURL), w, req)
+}
+
+func (s *Server) handleAppLivenessProbe(w http.ResponseWriter, req *http.Request) {
+	requestStatusCode(fmt.Sprintf("http://127.0.0.1%s", s.appLiveURL), w, req)
+}
+
+func requestStatusCode(appURL string, w http.ResponseWriter, req *http.Request) {
+	httpClient := &http.Client{
```

Comment:
Create an httpClient object one per probe.
This will create a new connection for every probe which is wasteful.

Reply:
Totally fine to change to reuse the same object, but not sure if that makes difference.

The object is cheap, the connection is not. If I understand code correctly, the httpClient itself does not maintain the connection. Instead the `DefaultRoundTripper` is used by default, go use the same Transport object to keep track of the connection.

[1] https://golang.org/pkg/net/http/#Client 
[2]https://godoc.org/github.com/elazarl/goproxy/transport#RoundTripper

Reply:
You are right `http.DefaultTransport` does cache and reuse connections.